integration/test_timescaledb.py

```diff
@@ -294,15 +286,375 @@ async def test_timescaledb_async_emit(timescaledb):
+        await controller.emit(data)
 
     await controller.terminate()
     await controller.await_termination()
 
     # Verify data was inserted
+    with connection.cursor() as cursor:
+        cursor.execute(f"SELECT COUNT(*) FROM {table_name}")
+        count = cursor.fetchone()[0]
+        assert count == 2, "Async emission should insert exactly 2 rows"
+
+
+@pytest.mark.parametrize("nullable", [True, False])
+def test_timescaledb_schema_validation_missing_column(timescaledb, nullable):
+    """Test that validation properly handles missing columns based on nullability"""
+    (connection, _, _, _, _, _, table_name, dsn_url, _, _) = timescaledb
+
+    # Create a table with nullable and non-nullable columns
+    cursor = connection.cursor()
+    validation_table = f"{table_name}_validation_{'nullable' if nullable else 'required'}"
+    cursor.execute(f"DROP TABLE IF EXISTS {validation_table};")
+
+    # Create table with different column nullability based on parameter
+    test_col_nullable = "NULL" if nullable else "NOT NULL"
+    cursor.execute(
+        f"""
+        CREATE TABLE {validation_table} (
+            time TIMESTAMPTZ NOT NULL,
+            required_col INTEGER NOT NULL,
+            test_col VARCHAR(50) {test_col_nullable}
+        );
+    """
+    )
+    cursor.execute(f"SELECT create_hypertable('{validation_table}', 'time');")
+    cursor.close()
+
+    time_format = "%d/%m/%y %H:%M:%S UTC%z"
+    controller = build_flow(
+        [
+            SyncEmitSource(),
+            TimescaleDBTarget(
+                dsn=dsn_url,
+                table=validation_table,
+                time_col="time",
+                columns=["required_col", "test_col"],
+                time_format=time_format,
+                max_events=10,
+            ),
+        ]
+    ).run()
+
+    # Test data missing the test column
+    test_data = {
+        "time": "18/09/19 01:55:10 UTC+0000",
+        "required_col": 42,
+        # Missing test_col
+    }
+
+    if nullable:
+        # Should work fine for nullable columns - missing column gets NULL
+        controller.emit(test_data)
+        controller.terminate()
+        controller.await_termination()
+
+        # Verify data was inserted with NULL for missing nullable column
+        with connection.cursor() as cursor:
+            cursor.execute(f"SELECT required_col, test_col FROM {validation_table}")
+            result = cursor.fetchone()
+            assert result[0] == 42
+            assert result[1] is None  # test_col should be None
+    else:
+        # Should raise ValueError for missing required (non-nullable) column
+        with pytest.raises(ValueError, match=r"Missing required non-nullable column 'test_col'"):
+            controller.emit(test_data)
+            controller.terminate()
+            controller.await_termination()
+
+    # Cleanup
+    with connection.cursor() as cursor:
+        cursor.execute(f"DROP TABLE IF EXISTS {validation_table} CASCADE;")
```

Comment:
The cleanup block will not run in case of a test failure.

Reply:
will fix